cli/start: enhance and better comment the startup process

knz opened the pull request:
Arjun and folk have expressed interest in better understanding
"what happens during CockroachDB start up". While studying the code
to prepare an explanation, I found out that the code was somewhat
under-documented, and the ordering of steps wasn't following
a reasonable order.

This patch enhances this situation by adding necessary comments
and tweaking the start up process:

- logging is set up earlier.
- configuration tweaks are grouped into a single packet
  of logic / statements.
- InitNode() is called in the main goroutine, earlier, so
  that configuration reporting and env var reporting can
  be done together.

Release note: None

cc @bdarnell

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/21344)
<!-- Reviewable:end -->

knz committed:
cli/start: enhance and better comment the startup process

Arjun and folk have expressed interest in better understanding
"what happens during CockroachDB start up". While studying the code
to prepare an explanation, I found out that the code was somewhat
under-documented, and the ordering of steps wasn't following
a reasonable order.

This patch enhances this situation by adding necessary comments
and tweaking the start up process:

- logging is set up earlier.
- configuration tweaks are grouped into a single packet
  of logic / statements.
- InitNode() is called in the main goroutine, earlier, so
  that configuration reporting and env var reporting can
  be done together.

Release note: None

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21344#-:-L2QdYgWEz53ypXjfpwh:bnfp4nl)*
<!-- Sent from Reviewable.io -->

knz commented:
TFYR!

knz committed:
cli/start: also warn when --max-sql-memory is not set

Prior to this patch, a warning was reported if the user
was running `cockroach` with the `--cache` parameter left
to its default value.

Such a warning is also useful for `--max-sql-memory`.
This patch adds it.

Release note: None
